### ISSUE TITLE
Count "where" parameter is not respected (it just counts all rows in the table with no filter)

### ISSUE THREAD

**Issue by JonathanWilbur** [prisma/prisma#11645]
### Bug description

This is my query:

```typescript
    const count = await ctx.db.enqueuedSearchResult.count({
        where: {
            connection_uuid: conn.id,
            query_ref: searchState.paging![0],
        },
    });
```

I would expect this to return the number of `EnqueuedSearchResult` having the specified `connection_uuid` AND `query_ref`. However, I see that it is simply ignoring the filter and querying all records in the table:

```SQL
SELECT COUNT(*) FROM (SELECT `directory`.`EnqueuedSearchResult`.`id` FROM `directory`.`EnqueuedSearchResult` WHERE 1=1) AS `sub`
```

Notice the `WHERE 1=1` above.

### How to reproduce

Use the `count()` API with a `where` selection that would, if executed as expected, produce fewer rows than the table contains. Observe that it always returns the total number of rows in the table.

**I suspect this is specific to MySQL. See the version numbers below.**

### Expected behavior

The `count()` API should respect the where clause and only return the number of entries that match the `where` filter, according same semantics by which `where` is evaluated with a `findMany()` query.

### Prisma information

I can post more if you want, but here's anything that I think would be relevant:

```prisma
datasource db {
  url      = env("DATABASE_URL")
  provider = "mysql"
  // provider = "postgresql"
  // provider = "sqlserver"
}

generator client {
  provider = "prisma-client-js"
}

// ...

model EnqueuedSearchResult {
  id              Int    @id @default(autoincrement())
  connection_uuid String @db.VarChar(40) // UUID of the connection.
  query_ref       String @db.VarChar(40) // Could be a pagination UUID or something else.
  result_index    Int // The 0-based index in which this entry appears in the results.
  entry_id        Int?
  entry           Entry? @relation(fields: [entry_id], references: [id], onDelete: Cascade)
  entry_info      Bytes // BER-encoded EntryInformation

  @@unique([connection_uuid, query_ref, result_index])
}
```

### Environment & setup

- OS: See `systeminfo` output below.
- Database:  MySQL Server 8.0.27
- Node.js version: v17.2.0

`systeminfo` output:

```
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19044 N/A Build 19044
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
```

### Prisma Version

```
prisma                  : 3.9.0
@prisma/client          : 3.9.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : bcc2ff906db47790ee902e7bbc76d7ffb1893009
Studio                  : 0.457.0
```

**Comment by JonathanWilbur** [comment]
The repo in which I found this issue is open-source, so you can just look at the entirety of where this happens. Link to the exact code location where I found this issue:

https://github.com/Wildboar-Software/directory/blob/7c37226dcc6f53eeb4b2376545c03556b5bc58c7/apps/meerkat/src/app/distributed/mergeSortAndPageSearch.ts#L390-L395

**Comment by pantharshit00** [comment]
Hello,

This issue was fixed in 3.9.1. I am going to close this as duplicate of https://github.com/prisma/prisma/issues/11556
